tests/GslCore.Tests/TestLineNumbers.fs

```diff
@@ -0,0 +1,161 @@
+
+    do
+        // initialize pragmas
+        pragmaTypes.finalizePragmas []
+
+    [<Test>]
+    member __.TestFunctionExpansion() =
+        let source = """#refgenome cenpk
+#platform stitch
+
+let knockout(up,down) =
+    &up ; &down
+end
+
+
+knockout(uADH1,dADH1)
+knockout(uADH2,dADH2)
+knockout(uADH3,dADH3)
+knockout(uADH4,dADH4)""" 
+
+        let result = 
+            source 
+            |> GslSourceCode
+            |> compile (phase1 Set.empty) 
+            |> returnOrFail
+
+
+        let assemblies = extractAssemblies result.wrappedNode
+
+        Assert.AreEqual(4,assemblies.Length)
+
+        // test coordinates of first assembly
+        let a1 = List.head assemblies
+        checkPosition a1.pos (8,0,8,8)
+
+        // test coordinates of last assembly
+        let a4 = List.item 3 assemblies
+        checkPosition a4.pos (11,0,11,8)
+
+    [<Test>]
+    member __.TestFunctionExpansionNoArgs() =
+        let source =    "#refgenome cenpk
+#platform stitch
+
+let knockout() =
+    uADH1 ; dADH1
+end
+knockout()" 
+                        
+        let result = 
+            source 
+            |> GslSourceCode
+            |> compile (phase1 Set.empty) 
+            |> returnOrFail
+
+        let assemblies = extractAssemblies result.wrappedNode
+
+        // test coordinates of first assembly
+        let a1 = List.head assemblies
+        checkPosition a1.pos (6,0,6,8)
+
+    [<Test>]
+    member i__.TestRecursiveFunctionExpansion() =
+        let source = """#refgenome cenpk
+#platform stitch
+
+let fun1(up,down) = // line 3 (zero numbered)
+&up ; &down // line 4
+end
+
+let funTwo(gene) = // line 7 
+fun1(&gene,&gene) // line 8
+end
+
+fun1(uADH1,dADH1) // line 11
+funTwo(uADH4) // line 12
+
+let functionThree() = // line 14
+ funTwo(uADH7) // line 15
+end
+
+
+         functionThree() // line 19
+""" 
+
+        let result = 
+            source 
+            |> GslSourceCode
+            |> compile (phase1 Set.empty) 
+            |> returnOrFail
+
+
+        let assemblies = extractAssemblies result.wrappedNode
+
+        // Helpful debugging tool
+        let _coordsFormatted =
+            String.Join(
+                "\n",
+                [for a in assemblies ->
+                        String.Join(";",
+                            [for p in a.positions ->
+                                sprintf "%d,%d->%d,%d" p.s.Line p.s.Column p.e.Line p.e.Column
+                            ]
+                        )
+                ]
+            )
+
+        Assert.AreEqual(3,assemblies.Length)
+
+        // test coordinates of first assembly
+        let a1 = List.head assemblies
+        checkPosition a1.pos (11,0,11,4)
+
+        // test coordinates of second assembly
+        let a2 = List.item 1 assemblies
+        checkPosition a2.pos (12,0,12,6)
+
+        // test coordinates of last assembly
+        let a3 = List.item 2 assemblies
+        checkPosition a3.pos (19,9,19,22)
```

Comment:
I'm assuming that this is the test that would catch the issue with missing nested function call positions?  Would love to fix this in the future.  Could you do the following:

1. File an issue that documents the problem, maybe with a GSL snippet that reproduces it here.
2. Add either a failing but skipped unit test, or some commented-out assertions in this test, that would pass if we fixed this problem.
3. Cross-reference the issue with the test.

Reply:
Done.  See issue #15 .  There is a test marked ignore that asserts correct behavior and currently fails.